Source/DataModels/Components/ComponentAnimation.cpp

```diff
@@ -51,10 +52,9 @@ void ComponentAnimation::SetStateMachine(const std::shared_ptr<ResourceStateMach
 
 void ComponentAnimation::Update()
 {
-
 	if (stateMachine)
 	{
-		if (actualState == 0) //Entry State 
+		if ((actualState == 0) & (lastState == NON_STATE)) //Entry State 
```

Comment:
I'm guessing you don't want a bit-wise comparison...
```suggestion
		if ((actualState == 0) && (lastState == NON_STATE))
```